### ISSUE TITLE
Signature grading improvements

### ISSUE THREAD

**Issue by Xyene** [DMOJ/judge#72]
The grader process should be separated from the submission process when signature grading. This prevents from all manner of exploits (not limited to `printf("Correct!\n"); exit(0);` and its numerous variants). In theory it should also increase grading speed, as the grader process would not have to be sandboxed.

The judge interfacing to the grader would not change, but the grader would interface with a submission through a piped protocol. If we limit graders to C/++ (and we should, but that's another issue), then the only work involved in adding a language for siggrading is writing the necessary client.

**tl;dr**
- No security issues
- Possibly faster grading
- Easier language support (trivial to add Java and Python; would allow usage of siggrading for DMOPCs etc.)

**Comment by quantum5** [comment]
Oh great, now have to implement piping grading, as if `judge.py` is actually understandable.

**Comment by Xyene** [comment]
Nowadays `judge.py` is much more understandable, so this is now a very real possibility.

**Comment by quantum5** [comment]
This is horribly misguided. Any benefit you can gain from running without sandbox will be destroyed by the I/O required for IPC, which is sandboxed at one end.
